### PR TITLE
Optimizing `<Image />` component

### PR DESCRIPTION
**Pull request by sanjaiyan-dev**
[Optimizing `<Image />` component](https://web.dev/browser-level-image-lazy-loading/).

1) Adding `decoding` attribute with `async` value which decode the image asynchronously to reduce delay in presenting other content.
2) Adding `loading` prop with `lazy` as a default value which defers loading the image until it reaches a calculated distance from the viewport, as defined by the browser. The intent is to avoid the network and storage bandwidth needed to handle the image until it's reasonably certain that it will be needed.

_Sorry if I made any mistakes :(_

### CODE REVIEW

**Comment by mportiz08** [comment]
@sanjaiyan-dev could I request two small changes to help me more effectively review this next week?

1. Try to keep the diff minimal by leaving syntax/whitespace-only changes to a minimum to make it easier to see your code changes (example: changing single quoted string to double quoted one, etc)

2. Can you add a quick overview to the PR that explains the optimizations that these changes introduce?

Thanks! Excited to see your contributions :)

**Comment by sanjaiyan-dev** [comment]
> [Optimizing `Image` component](https://web.dev/lazy-loading-images/).
> 
> _Sorry if I made any mistakes :(_



> @sanjaiyan-dev could I request two small changes to help me more effectively review this next week?
> 
> 1. Try to keep the diff minimal by leaving syntax/whitespace-only changes to a minimum to make it easier to see your code changes (example: changing single quoted string to double quoted one, etc)
> 2. Can you add a quick overview to the PR that explains the optimizations that these changes introduce?
> 
> Thanks! Excited to see your contributions :)

Ok 👍🏽

**Comment by sanjaiyan-dev** [comment]
> @sanjaiyan-dev could I request two small changes to help me more effectively review this next week?
> 
> 1. Try to keep the diff minimal by leaving syntax/whitespace-only changes to a minimum to make it easier to see your code changes (example: changing single quoted string to double quoted one, etc)
> 2. Can you add a quick overview to the PR that explains the optimizations that these changes introduce?
> 
> Thanks! Excited to see your contributions :)

@mportiz08, Done 🙌👍

File: src/components/ImageAsset.vue [hunk 1]
```diff
@@ -129,6 +134,10 @@ export default {
       type: Array,
       required: true,
     },
+    loading: {
+      type: String,
+      default: 'lazy'
```
Comment (mportiz08) [inline src/components/ImageAsset.vue:139]:
```suggestion
      default: 'lazy',
```

(Super tiny thing, but the linter is complaining about this)

**Review by mportiz08 — APPROVED** [review]
Awesome! I think these changes will work great as defaults for this component and shouldn't cause issues with older versions of browsers that don't support lazy loading of images yet.

Thanks @sanjaiyan-dev!

**Comment by sanjaiyan-dev** [comment]
> @sanjaiyan-dev One additional thing I just thought of: could we update the unit tests to check for the presence of these new attributes? I think it could be as simple as adding a single assertion to an existing test in `tests/unit/components/ImageAsset.spec.js`.

Sorry, I have added test (not sure if it is correct :( ) , can you please help me with it 🙇🏽

File: tests/unit/components/ImageAsset.spec.js [hunk 2]
```diff
@@ -54,6 +54,7 @@ describe('ImageAsset', () => {
     expect(image.attributes('width')).toBe('1202');
     expect(image.attributes('height')).toBe('auto');
     expect(image.attributes('alt')).toBe(alt);
+    expect(image.attributes('decoding').toBe('async'));
```
Comment (mportiz08) [inline tests/unit/components/ImageAsset.spec.js:57]:
```suggestion
    expect(image.attributes('decoding')).toBe('async');
```

File: tests/unit/components/ImageAsset.spec.js [hunk 3]
```diff
@@ -85,6 +86,7 @@ describe('ImageAsset', () => {
     expect(image.attributes('width')).toBe('300');
     expect(image.attributes('height')).toBe('auto');
     expect(image.attributes('alt')).toBe('');
+    expect(image.attributes('decoding').toBe('async'));
```
Comment (mportiz08) [inline tests/unit/components/ImageAsset.spec.js:89]:
```suggestion
    expect(image.attributes('decoding')).toBe('async');
```

File: tests/unit/components/ImageAsset.spec.js [hunk 4]
```diff
@@ -128,6 +130,7 @@ describe('ImageAsset', () => {
     expect(image.attributes('srcset')).toBe(`${url2x} 2x, ${url3x} 3x`);
     expect(image.attributes('width')).toBe('1202');
     expect(image.attributes('height')).toBe('auto');
+    expect(image.attributes('decoding').toBe('async'));
```
Comment (mportiz08) [inline tests/unit/components/ImageAsset.spec.js:133]:
```suggestion
    expect(image.attributes('decoding')).toBe('async');
```

**Review by mportiz08 — CHANGES_REQUESTED** [review]
> Sorry, I have added test (not sure if it is correct :( ) , can you please help me with it 🙇🏽

Sure, no problem. I think there are just some minor syntax issues with the assertions (see suggested changes).

Also, there are now some failing tests from `tests/unit/components/Asset.spec.js` that I think could be easily addressed with the following change:

```diff
diff --git a/tests/unit/components/Asset.spec.js b/tests/unit/components/Asset.spec.js
index 8e7bc18..59d1c54 100644
--- a/tests/unit/components/Asset.spec.js
+++ b/tests/unit/components/Asset.spec.js
@@ -185,6 +185,7 @@ describe('Asset', () => {
       expect(asset.exists()).toBe(true);
       expect(asset.props()).toEqual({
         alt: image.alt,
+        loading: "lazy",
         variants: image.variants,
       });
     });
@@ -194,6 +195,7 @@ describe('Asset', () => {
       const asset = wrapper.find(ImageAsset);
       expect(asset.props()).toEqual({
         alt: image.alt,
+        loading: "lazy",
         variants: image.variants,
       });
     });
```

Thanks again for helping with this!

**Comment by sanjaiyan-dev** [comment]
🤔

**Review by mportiz08 — APPROVED** [review]
🚀

**Comment by mportiz08** [comment]
@swift-ci test

**Comment by mportiz08** [comment]
Thanks @sanjaiyan-dev — I think this will have a noticeable improvement on the rendering performance! :)

**Comment by sanjaiyan-dev** [comment]
Tks a lot 🎉

**Comment by sanjaiyan-dev** [comment]
> Thanks @sanjaiyan-dev — I think this will have a noticeable improvement on the rendering performance! :)

Tks a lot 🎉

**Pull request merged**